src/coreclr/jit/vartype.h

```diff
@@ -61,30 +62,16 @@ inline var_types TypeGet(var_types v)
     return v;
 }
 
-#ifdef FEATURE_SIMD
-template <class T>
-inline bool varTypeIsSIMD(T vt)
-{
-    switch (TypeGet(vt))
-    {
-        case TYP_SIMD8:
-        case TYP_SIMD12:
-        case TYP_SIMD16:
-        case TYP_SIMD32:
-            return true;
-        default:
-            return false;
-    }
-}
-#else  // FEATURE_SIMD
-
-// Always return false if FEATURE_SIMD is not enabled
 template <class T>
 inline bool varTypeIsSIMD(T vt)
 {
+#ifdef FEATURE_SIMD
+    return ((varTypeClassification[TypeGet(vt)] & VTF_VEC) != 0);
```

Comment:
An alternative would be: `return (TypeGet(vt) >= TYP_SIMD8) && (TypeGet(vt) <= TYP_SIMD32);`

I'd expect `varTypeClassification` to typically be cached though, so I'm operating under the assumption a new classification is "better".